source/parser.hera

```diff
@@ -3465,12 +3465,14 @@ Yield
 # https://facebook.github.io/jsx/#prod-JSXElement
 JSXElement
   JSXSelfClosingElement
-  JSXOpeningElement JSXChildren* __ JSXClosingElement ->
+  JSXOpeningElement JSXChildren? __ JSXClosingElement ->
     // Check that tags match
     if ($1[1] !== $4[2]) {
       throw new Error(`mismatched closing tags at ${JSON.stringify($loc)}`)
     }
     return $0
+  JSXOpeningElement:open JSXNestedChildren InsertNewline InsertIndent ->
```

Comment:
I would probably move this rule above the non-indented rule. It shouldn't really matter but with caching and the parser state tracking sometimes it does.

Reply:
Oddly, this fixed the `InsertIndent` problem. Now I get:
```js
return
  <div>
    <b>
      Bold
    </b>
    <i>
      Italic
    </i>
</div>
console.log('hi')
```
which seems about right.

I had in mind that it's good to check for an explicit closing tag first, but I'm not sure what's best from a performance perspective. What's weird is that it affects correctness of `InsertIndent`...

Reply:
This change also breaks regular JSX, specifically the "nested tags" example.
```jsx
    <p>
      <a href="https://www.reactjs.org"> link </a>
      <img src={user.avatarUrl}></img>
    </p>
```
I guess indentation-based and explicitly closed tags aren't so easy to make co-exist...